app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt

```diff
@@ -0,0 +1,913 @@
+      onView(withId(R.id.extra_controls_title))
+        .check(matches(withText(R.string.administrator_controls_edit_profiles)))
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_defaultTabletConfig_profileListIsDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      )
+      onView(
+        atPositionOnView(
+          R.id.profile_list_recycler_view,
+          0,
+          R.id.profile_list_admin_text
+        )
+      ).check(
+        matches(withText(context.resources.getString(R.string.profile_chooser_admin)))
+      )
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name)).check(
+        matches(withText("Ben"))
+      )
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_tabletConfigChange_profileListIsDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      )
+      onView(
+        atPositionOnView(
+          R.id.profile_list_recycler_view,
+          0,
+          R.id.profile_list_admin_text
+        )
+      ).check(
+        matches(withText(context.resources.getString(R.string.profile_chooser_admin)))
+      )
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name)).check(
+        matches(withText("Ben"))
+      )
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_selectProfileAdmin_backButton_selectSecondProfileDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.extra_controls_title)).check(matches(withText("Admin")))
+      onView(withId(R.id.profile_edit_name)).check(matches(withText("Admin")))
+      onView(withId(R.id.administrator_controls_multipane_options_back_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name)).check(
```

Comment:
this line is unable to execute therefore it is causing this test to fail.

Reply:
Can you be more specific on what 'unable to execute' means? What specifically are you observing?

Reply:
like at line 683 a click action is performed but it is actually not happening so the test fails as this action is required.

Reply:
as this click is very dependent as it loads the ProfielEditFragment so the checks in furthur lines fails and therefore the test fails